src/main/java/org/fiware/iam/tir/rest/TrustedIssuersRegistry.java

```diff
@@ -0,0 +1,54 @@
+package org.fiware.iam.tir.rest;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.fiware.iam.tir.api.TirApi;
+import org.fiware.iam.tir.issuers.IssuersProvider;
+import org.fiware.iam.tir.issuers.TrustedIssuerMapper;
+import org.fiware.iam.tir.model.IssuerVO;
+import org.fiware.iam.tir.model.IssuersResponseVO;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+@RequiredArgsConstructor
+@Slf4j
+@Controller("${general.basepath:/}")
+@Secured(SecurityRule.IS_ANONYMOUS)
+public class TrustedIssuersRegistry implements TirApi {
+    private final IssuersProvider issuersProvider;
+    private final TrustedIssuerMapper mapper;
+
+    @Override
```

Comment:
Maybe I missed it somewhere, but since you are using an async repository but return a sync HttpResponse, where do you block(and why;))? There is a real risk to get timeouts, because of micronaut killing the blocked requests under the hood(seen this much to often;))

Reply:
Gonna add an issue to checkafterwards